src/CoreBundle/private/library/classes/TCMSTableEditor/TCMSTableFieldWriter.class.php

```diff
@@ -120,66 +136,118 @@ public function Save(&$postData, $bDataIsInSQLForm = false)
             return false;
         }
 
-        $bTargetTableChangeForMLTField = false;
-        if (array_key_exists('bTargetTableChangeForMLTField', $postData)) {
-            $bTargetTableChangeForMLTField = $postData['bTargetTableChangeForMLTField'];
-        }
-        // if the name of the field has changed we need to alter the table definition
-        // we also need to change the field definition if the type changed
-        $bTypeChanged = false;
-        if (array_key_exists('cms_field_type_id', $this->oTable->sqlData)) {
-            $bTypeChanged = ($this->oTable->sqlData['cms_field_type_id'] != $postData['cms_field_type_id']);
-            $currentFieldTypeQuery = "SELECT * FROM `cms_field_type` WHERE `id` = '".MySqlLegacySupport::getInstance()->real_escape_string($this->oTable->sqlData['cms_field_type_id'])."'";
-            $currentFieldTypeResult = MySqlLegacySupport::getInstance()->query($currentFieldTypeQuery);
-            $currentFieldTypeRow = MySqlLegacySupport::getInstance()->fetch_assoc($currentFieldTypeResult);
-        }
-        $aOriginalData = $this->oTable->sqlData;
+        $this->oldData = $this->oTable->sqlData;
+        $this->oldTableName = $this->_oParentRecord->sqlData['name'];
+        $this->isChangeFromTable = $postData['bTargetTableChangeForMLTField'] ?? false; // true (from TCMSTableWriter) prohibits target changes
 
-        $newFieldTypeQuery = "SELECT * FROM `cms_field_type` WHERE `id` = '".MySqlLegacySupport::getInstance()->real_escape_string($postData['cms_field_type_id'])."'";
-        $newFieldTypeResult = MySqlLegacySupport::getInstance()->query($newFieldTypeQuery);
-        $newFieldTypeRow = MySqlLegacySupport::getInstance()->fetch_assoc($newFieldTypeResult);
+        return parent::Save($postData);
+    }
 
-        $oldName = $postData['name'];
-        if (array_key_exists('name', $this->oTable->sqlData)) {
-            $oldName = $this->oTable->sqlData['name'];
-        }
+    /**
+     * {@inheritdoc}
+     */
+    protected function PostSaveHook(&$oFields, &$oPostTable)
+    {
+        parent::PostSaveHook($oFields, $oPostTable);
+
+        $this->_oField = $this->adaptFieldAndRelatedTables($oPostTable->sqlData);
 
-        if ($bTypeChanged) {
-            $this->_oField->ChangeFieldTypePreHook();
+        $this->getAutoclassesCacheWarmer()->updateTableById($oPostTable->sqlData['cms_tbl_conf_id']);
+    }
+
+    /**
+     * @param array $postData
+     *
+     * @return TCMSField - the new definition after changes
+     *
+     * @throws DBALException
+     */
+    protected function adaptFieldAndRelatedTables(array $postData): TCMSField
+    {
+        $oldTypeId = $this->oldData['cms_field_type_id'];
+        $newTypeId = $postData['cms_field_type_id'];
+        $newName = $this->oTable->sqlData['name'];
+
+        $bTypeChanged = $oldTypeId !== $newTypeId;
+        $currentFieldTypeRow = $this->getFieldTypeDefinition($oldTypeId);
```

Comment:
I'd use the prefix `old` instead of `current` to for consistency.

Reply:
This has not been addressed yet.

Reply:
ok